src/components/Security/Profiles/Filters.vue

```diff
@@ -1,79 +1,135 @@
 <template>
-  <div class="Filters">
-    <div class="row card-panel card-header">
-      <div class="col s7">
-        <role-chips
-          :added-roles="addedRoles"
-          @selected-role="onRoleSelected"
-          @remove-role="removeRole"
-        />
-      </div>
-      <div class="Filters-actions col s3">
-        <button
-          type="submit"
-          class="btn btn-small waves-effect waves-light"
-          @click.prevent="submitSearch"
-        >
-          {{ labelSearchButton }}
-        </button>
-        <button
-          class="btn-flat btn-small waves-effect waves-light"
-          @click="reset"
-        >
-          Reset
-        </button>
-      </div>
-    </div>
-  </div>
+  <b-card no-body data-cy="ProfileFilters">
+    <template v-slot:header>
+      <b-row>
+        <b-col cols="9" class="vertical-align">
+          Search by role
+          <b-dropdown
+            class="ml-2"
+            data-cy="ProfileFilters-roleSelect"
+            variant="outline-primary"
+            menu-class="dropdownScroll"
+            text="Select roles to be contained in the profiles"
+            no-flip
+          >
+            <b-dropdown-text v-for="role of roleList" :key="`dropdown-${role}`">
+              <div
+                class="inlineDisplay pointer"
+                :data-cy="`RoleSelect--${role}`"
+              >
+                <span class="inlineDisplay-item">
+                  <b-form-checkbox
+                    class="mx-2"
+                    :checked="roleIsSelected(role)"
+                    :id="role"
+                    @change="toggleRole(role, $event)"
+                  />
+                </span>
+                <label
+                  class="inlineDisplay-item code pointer"
+                  :for="role"
+                  :title="role"
+                  >{{ truncateName(role, 20) }}</label
+                >
+              </div>
+            </b-dropdown-text>
+            <b-dropdown-text v-if="roleList.length === 0">
+              <span class="inlineDisplay-item">
+                No roles found.
+              </span>
+            </b-dropdown-text>
+          </b-dropdown>
+          <b-badge
+            v-if="hasFilter"
+            data-cy="ProfileFilters-filterAppliedPill"
+            pill
+            variant="info"
+            class="ml-2 py-2 px-3"
+            >Filters are being applied</b-badge
+          >
+        </b-col>
+        <b-col class="text-right">
+          <b-button
+            class="mr-2"
+            data-cy="ProfileFilters-resetBtn"
+            variant="outline-primary"
+            @click="reset"
+          >
+            Reset
+          </b-button>
+          <b-button
+            type="submit"
+            variant="primary"
+            @click.prevent="submitSearch"
+          >
+            {{ labelSearchButton }}
+          </b-button>
```

Comment:
I'm not sure about the necessity of this button. 
When we add a role in the selector, the list is automatically updated. 
This search button just use the router to push on the same page with the same filters so that trigger a `navigationDuplicated` error.
If we decided to keep that button to refresh the list displayed, we might need to change the way the list works (using the router).
But IMHO, it doesn't make sens because the list is always updated with the last role filters we setted.

Reply:
OK let's take it away